src/classrooms/classrooms.service.ts

```diff
@@ -65,4 +65,17 @@ export class ClassroomsService {
     // TODO: query for fetching classrooms that a student is enrolled in.
     return classrooms;
   }
+
+  async getClassroomById(id: number, userId: number): Promise<Classroom | null> {
+    const user = await this.em.findOneOrFail(User, { id: userId }, { populate: ["teacher"] });
```

Comment:
```suggestion
    const user = await this.usersRepository.findOneOrFail({ id: userId }, { populate: ["teacher"] });
```

Same comment for other repetitions of this in this file.